pkg/verification/verify.go

```diff
@@ -44,6 +166,36 @@ func VerifyTimestampResponse(tsrBytes []byte, artifact io.Reader, certPool *x509
 		return err
 	}
 
+	err = verifyNonce(ts.Nonce, opts)
+	if err != nil {
+		return err
+	}
+
+	err = verifyOID(ts.Policy, opts)
+	if err != nil {
+		return err
+	}
+
+	err = verifyEmbeddedLeafCert(ts.Certificates[0], opts)
+	if err != nil {
+		return err
+	}
+
+	err = verifyLeafAndIntermediatesEKU(opts)
+	if err != nil {
+		return err
+	}
+
+	err = verifyLeafCertSubject(ts.Certificates[0].Subject.String(), opts)
```

Comment:
Question: I am not sure the response will always contain certificates. What if the request sets `Certificates: false` ? I am just wondering it so we avoid any panic.

Reply:
If the request has `Certificates: false`, maybe we could ignore the leaf cert verifications. @haydentherapper do you have any thoughts around this?

Reply:
If `Certificates: false`, then the leaf must be provided out of band. Either:
* You give us the certificate chain (intermediates + root), and the leaf comes from the response
* You give us the leaf + chain, and the response has no leaf
* You give us the leaf + chain, and the response has a leaf - In this case, we should just confirm that the leaves match imo